tools/cli/datalab.py

```diff
@@ -224,7 +224,8 @@ def run():
         _SUBCOMMANDS[args.subcommand]['run'](
             args, gcloud_compute,
             gcloud_repos=gcloud_repos,
-            email=get_email_address())
+            email=get_email_address(),
+            in_cloud_shell=('DEVSHELL_CLIENT_PORT' in os.environ))
```

Comment:
> is using an environment var reliable enough? 

I think it is.

This seems to be the established pattern for detecting running in the Cloud Shell.

In particular, this is the pattern that is followed by the [oauth2client](https://github.com/google/oauth2client/blob/master/oauth2client/contrib/devshell.py) library.

The corresponding concept in the oauth2client library would be raising a 'NoDevshellServer' error, and that only gets raised if the environment variable is not set.